demo/src/jvmMain/java/main.kt

```diff
@@ -1,12 +1,26 @@
 package demo
 
 import drewcarlson.qbittorrent.QBittorrentClient
+import io.ktor.client.*
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.runBlocking
+import java.net.InetSocketAddress
+import java.net.Proxy
 
 fun main(vararg args: String) = runBlocking {
     val client = QBittorrentClient(
-        baseUrl = args.firstOrNull() ?: "http://localhost:9090"
+        baseUrl = "http://192.168.88.6:8080",
+        httpClient = HttpClient {
+            engine {
+                proxy = Proxy(Proxy.Type.HTTP, InetSocketAddress(8888))
+            }
+        }
     )
 
+    client.syncMainData()
```

Comment:
Print each `mainData` object and move the version `println` above here.

Reply:
qBittorrent is not sending valules that are not changed so it writes nulls sometimes. Should I save the state and keep updating it on new requests? so that it will always be non-null

![image](https://user-images.githubusercontent.com/22417494/110554678-b511a300-8154-11eb-8f56-63c4c3d1d05d.png)

Reply:
Ah, right!  That change would be very much appreciated, let me know if you'd like to do it now or as a separate PR and I can merge this first.

If you need it, `torrentFlow` is an example of what you want to do:
https://github.com/DrewCarlson/qBittorrent-Kotlin/blob/db5c6656b60fc2fc5b18558085c91b4bebb865c5/client/src/commonMain/kotlin/QBittorrentClient.kt#L138-L140

Reply:
god damn this was a lot of copy pase, but it works :D
![image](https://user-images.githubusercontent.com/22417494/111545757-f8d95d80-878f-11eb-8517-82d80631fd14.png)

I have added MainDataDTO, which has all the nullable variables and then there is MainData, which doesnot have any nullable values :D
 same is for ServerState
 
and the end result is that there are no nulls being printed anywhere, nor the need to use nullchecks

Reply:
I hav few other ideas but I will write them tomorrow, gotta sleep now..

Reply:
okay, it is all working but `fun torrentFlow(hash: String)` needs some refactoring, because  it requests `/api/v2/torrents/info` for torrent data when it already has the data from previous requests. This is also going to need a separate TorrentDTO model and then a lot of null checks.... and im lazy to refactor all that.

anyways it is fine.... I guess...

Reply:
hello? are you here?

Reply:
Thank you, I'll finish this and merge when I have time.

Reply:
okay, its been a while. I think I will add another static models for torrentFlow. I plan to do similar things for Torrent class as I did with ServerState class. 

will that be ok? or maybe you already started to work on this? Im gonna have lots of free time this weekend...

Reply:
I have not had a chance to work on this yet.  The extra models are not required for ServerState or Torrent.  The initial ServerState object can be stored as a `JsonObject` and have updates applied directly to the json representation.  For each emission the `JsonObject` serialized into a complete object even though the response only contained some fields.

Reply:
I would disagree. There should be DTOs which will have nullable types and be Serializable, then mapped to other models that will not be nullable or Serializable. Otherwise there would be so many random strings to search in JsonObject, and all that without auto completion...

Reply:
Thanks again.  I've merged the ServerState model here https://github.com/DrewCarlson/qBittorrent-Kotlin/pull/5/commits/f573d2fd24fa1d427afadd22bdb16ca2e6071dbb